lib/solr.ts

```diff
@@ -150,9 +153,9 @@ export class Client {
     const request = pickProtocol(this.options.secure).request(requestOptions);
     request.on(
       'response',
-      handleJSONResponse(request, this.options.bigint, callback)
+      handleJSONResponse(request, this.options.bigint, callback, this.logger)
     );
-    request.on('error', function onError(err) {
+    request.on('error', (err) => {
```

Comment:
Since I'm working on requests now, we unfortunately bumped into each other again. Sorry!

Reply:
I suggest we merge current implementation as-is, you ignore my changes on request and plug the logger in wherever it will make sense to plug after you are done.

Reply:
Hmm, OK. "Where to plug in a logger" is actually quite subjective. I personally prefer letting errors bubble up and logging them with a global handler (which is part of an application, not a library). So I think I should leave that to you.

Reply:
Fair, we might actually not even need a logging solution after we go full async; this PR was created out of frustration our SRE experienced today when he had to debug existing solution, which generates no useful errors at all, since current SolrError conversion swallows all useful information about request failure. If we get rid of callback complexity and retain original error, that should actually suffice for majority or all cases. So I'll close PR for now, let's reevaluate if we still need it after you are done with your part.